server-common/src/main/java/org/apache/gravitino/server/authorization/MetadataFilterHelper.java

```diff
@@ -87,19 +101,45 @@ public static NameIdentifier[] filterByExpression(
       String expression,
       Entity.EntityType entityType,
       NameIdentifier[] nameIdentifiers) {
-    AuthorizationExpressionEvaluator authorizationExpressionEvaluator =
-        new AuthorizationExpressionEvaluator(expression);
     if (!enableAuthorization()) {
       return nameIdentifiers;
     }
-    return Arrays.stream(nameIdentifiers)
-        .filter(
-            metaDataName -> {
-              Map<Entity.EntityType, NameIdentifier> nameIdentifierMap =
-                  spiltMetadataNames(metalake, entityType, metaDataName);
-              return authorizationExpressionEvaluator.evaluate(nameIdentifierMap);
-            })
-        .toArray(NameIdentifier[]::new);
+    checkExecutor();
+    return RequestAuthorizationCache.executeWithThreadCache(
```

Comment:
ThreadLocal is hard to use in the thread pool. Because it may use wrong thread local.